### PR TITLE
I5pranay93/infws 145 ux asper mockups

### PR DESCRIPTION
**Pull request by i5pranay93**
## Description
Adds UI to show repository view as per mockups. Adds tabular view to show cookbook and it's recipes

### CODE REVIEW

File: src/dashboard/index.js [hunk 1]
```diff
@@ -37,6 +38,230 @@ function render() {
...
   });
+
+    document.getElementById("activity").addEventListener("click", () => {
+        const repository = path.join("file://", __dirname, "activity.html");
+        let os = require("os");
+        fetch(repository)
+            .then((response) => response.text())
+            .then((text) => {
+                document.querySelector("#content").innerHTML = text;
+                var cookbook_activity = document.createElement('div');
```
Comment (sanjain-progress) [inline src/dashboard/index.js:49]: any specific reason why we use `var` instead of `const`

File: src/dashboard/index.js [hunk 2]
```diff
@@ -37,6 +38,230 @@ function render() {
...
       });
   });
+
+    document.getElementById("activity").addEventListener("click", () => {
+        const repository = path.join("file://", __dirname, "activity.html");
+        let os = require("os");
+        fetch(repository)
+            .then((response) => response.text())
+            .then((text) => {
+                document.querySelector("#content").innerHTML = text;
```
Comment (nikhil2611) [inline src/dashboard/index.js:48]: where this content id is defined ?

File: src/dashboard/index.js [hunk 3]
```diff
@@ -37,6 +38,230 @@ function render() {
...
+                var cookbook_activity = document.createElement('div');
+                cookbook_activity.setAttribute("id", "cookbook-activity");
+
+                var container_cookbook_details = document.createElement('div');
```
Comment (nikhil2611) [inline src/dashboard/index.js:52]: we could use let here

File: src/dashboard/index.js [hunk 4]
```diff
@@ -37,6 +38,230 @@ function render() {
...
+                container_cookbook_details.setAttribute("class", "container");
+
+                var cookbook_details = document.createElement('div');
```
Comment (nikhil2611) [inline src/dashboard/index.js:55]: here too let could be used and in other declarations also

File: src/dashboard/index.js [hunk 5]
```diff
@@ -37,6 +38,230 @@ function render() {
...
+
+                    var td1 = document.createElement('td');
+
+                    var input_1 =  document.createElement('input'); //input form
+                    input_1.setAttribute("type", "checkbox"); // this will be set based on input
+                    input_1.setAttribute("class", "checkthis");
+
+                    // check if it is is checked
+
+                    var td2 = document.createElement('td');
```
Comment (nikhil2611) [inline src/dashboard/index.js:195]: This could be handled dynamically

**Comment by vkarve-chef** [comment]
general inputs & questions
- we can remove mockup content where functionality isn't yet available i.e. "Recent Updates" in header, search bar
- we can hide columns - "Recipes (count)" and "Created At" 
- while scrolling to the bottom, the header content and left nav moves up. can we avoid that?
- are long cookbook names truncated with alt text indicating complete name?

File: src/dashboard/index.js (hunk 1 above)
Comment (i5pranay93) [reply src/dashboard/index.js:49]: @sanjain-progress  const object properties cannot be changed, but i am changing it by setting attributes and updating them by appending the child object.

File: src/dashboard/index.js (hunk 2 above)
Comment (i5pranay93) [reply src/dashboard/index.js:48]: https://github.com/chef/chef-workstation-app/blob/73f488b8283855b412222aa86001e2abd7d00415/src/dashboard/index.html#L32

Pretty much similar to this - https://github.com/chef/chef-workstation-app/blob/73f488b8283855b412222aa86001e2abd7d00415/src/dashboard/index.js#L13

File: src/dashboard/index.js (hunk 3 above)
Comment (i5pranay93) [reply src/dashboard/index.js:52]: Sure we can, they have an almost similar property, and let is es6 property (as it hardly made any difference I used var)

File: src/dashboard/index.js (hunk 5 above)
Comment (i5pranay93) [reply src/dashboard/index.js:195]: We can definitely optimized it through the loop, I had given it a thought, 
e.g
 ```
for (var i = 1; i <7; i++) {
   var tdi =  document.createElement('td');
}
```

it was having some issue, will try and optimize it again.

**Pull request merged**